bin/opt8n/src/opt8n.rs

```diff
@@ -213,29 +214,17 @@ impl Opt8n {
             .ok_or_else(|| eyre!("Failed to create AlloyDB"))?,
         );
 
-        let block_env = BlockEnv {
-            number: U256::from(block.header.number),
-            coinbase: block.header.beneficiary,
-            timestamp: U256::from(block.header.timestamp),
-            difficulty: block.header.difficulty,
-            gas_limit: U256::from(block.header.gas_limit),
-            prevrandao: Some(block.header.mix_hash),
-            basefee: U256::from(block.header.base_fee_per_gas.unwrap_or_default()),
-            blob_excess_gas_and_price: block
-                .header
-                .excess_blob_gas
-                .map(|excess_gas| BlobExcessGasAndPrice::new(excess_gas as u64)),
-        };
-
-        let mut evm = EvmBuilder::default()
-            .with_db(Box::new(revm_db))
-            .with_block_env(block_env)
-            .build();
+        let mut evm = evm(
+            block,
+            self.eth_api.chain_id(),
+            CacheDB::new(revm_db),
+            SpecId::from(self.node_config.hardfork.unwrap_or_default()),
+        );
 
-        evm.context.evm.env.cfg.chain_id = self.eth_api.chain_id();
         for tx in block.transactions.iter() {
             let pending = PendingTransaction::new(tx.clone().into())?;
-            evm.context.evm.env.tx = pending.to_revm_tx_env();
+            let mut tx_env = pending.to_revm_tx_env();
+            tx_env.optimism.enveloped_tx = Some(Bytes::default());
```

Comment:
I'm surprised this works, shouldn't we need to actually encode the tx and provide it as the `enveloped_tx`?